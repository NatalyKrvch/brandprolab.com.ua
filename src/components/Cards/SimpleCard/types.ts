export interface SimpleCardProps {
  borderColor?: string;
  className?: string;
  backgroundColor?: string;
  backgroundImageUrl?: string;
}
