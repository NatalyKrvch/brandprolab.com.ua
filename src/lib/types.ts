export type Size = 's' | 'm' | 'l';
export type Screen = 'mobile' | 'tablet' | 'desktop';
