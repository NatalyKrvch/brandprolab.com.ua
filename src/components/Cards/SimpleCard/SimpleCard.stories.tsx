import type { Meta, StoryObj } from '@storybook/react';
import SimpleCard from './SimpleCard';

const meta: Meta<typeof SimpleCard> = {
  title: 'Components/SimpleCard',
  component: SimpleCard,
  parameters: {
    controls: { expanded: true },
  },
  argTypes: {
    children: {
      control: 'text',
      description: 'The content inside the card.',
      table: {
        type: { summary: 'ReactNode' },
      },
    },
    className: {
      control: 'text',
      description: 'Additional classes for styling.',
      table: {
        type: { summary: 'string' },
      },
    },
    backgroundColor: {
      control: { type: 'select' },
      options: ['bg-white', 'bg-teal-dark', 'bg-gray-light', 'bg-red-default'],
      description: 'Background color of the card.',
      table: {
        type: { summary: 'string' },
      },
    },
  },
};

export default meta;
type Story = StoryObj<typeof SimpleCard>;

export const ColoredCard: Story = {
  args: {
    children: 'This card has a colored background.',
    backgroundColor: 'bg-teal-dark',
    className: 'p-6 text-white',
  },
};
