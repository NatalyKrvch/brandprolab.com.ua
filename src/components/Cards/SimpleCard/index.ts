export { default as SimpleCard } from './SimpleCard';
export * from './types';
