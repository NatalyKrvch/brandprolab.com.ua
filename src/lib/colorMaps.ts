import {
  MainSocialCardColor,
  socialMediaColorClassMapType,
} from '@/components/Cards/SocialMediaCard/types';

export const socialMediaColorClassMap: socialMediaColorClassMapType = {
  [MainSocialCardColor.RED]: {
    bg: 'bg-red-default',
    text: 'text-red-default',
  },
  [MainSocialCardColor.BLUE]: {
    bg: 'bg-blue-default',
    text: 'text-blue-default',
  },
};
