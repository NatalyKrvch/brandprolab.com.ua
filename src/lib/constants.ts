export const ALL_DATA_QUERY = `
{
  "header": *[_type == "header"][0],
  "hero": *[_type == "hero"][0],
  "cases": *[_type == "cases"][0],
  "services": *[_type == "services"][0],
  "testimonials": *[_type == "testimonialsSection"][0],
  "subscribe": *[_type == "subscribeSection"][0],
  "video": *[_type == "videoSection"][0],
  "expertise": *[_type == "expertiseSection"][0],
  "diagnostic": *[_type == "diagnosticSection"][0],
  "footer": *[_type == "footerSection"][0]
}
`;

export const YOUTUBE_REGEX =
  /(?:youtu\.be\/|youtube\.com\/(?:.*[?&]v=|embed\/|v\/|shorts\/|live\/))([^&?/]+)/;

export enum IconClassType {
  SERVICES = 'services',
  EXPERTISE = 'expertise',
  SOCIAL_MEDIA = 'social-media',
  SOCIAL_MEDIA_MAIN = 'social-media-main',
}
export const REVALIDATE = 60;
export const ENDLESS_CAROUSEL_SPEED = 50;
export const SLIDER_RIGHT_SHIFT = 2;
export const SLIDER_LEFT_SHIFT = -1;
export const AMOUNT_OF_TESTIMONIALS_WORDS = 15;
export const AMOUNT_OF_CASES_WORDS_DESKTOP = 13;
export const AMOUNT_OF_CASES_WORDS_TABLET = 9;
export const AMOUNT_OF_CASES_WORDS_MOBILE = 13;
