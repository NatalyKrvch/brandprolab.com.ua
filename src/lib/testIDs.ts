export const INTERNAL_LINK_TEST_ID = 'internal-link-test-id';
export const EXTERNAL_LINK_TEST_ID = 'external-link-test-id';
export const FULL_WIDTH_CONTAINER_TEST_ID = 'full-width-container-test-id';
export const LIMITED_WIDTH_CONTAINER_TEST_ID =
  'limited-width-container-test-id';
export const LEFT_LIMITED_CONTAINER_TEST_ID = 'left-limited-container-test-id';
export const ICON_COMPONENT_TEST_ID = 'icon-component-test-id';
export const ICON_COMPONENT_IMAGE_TEST_ID = 'icon-component-image-test-id';
export const LABEL_COMPONENT_TEST_ID = 'label-component-test-id';
export const BUTTON_TEST_ID = 'button-test-id';
export const UNDERLINED_TEXT_TEST_ID = 'underlined-text-test-id';
export const USER_BADGE_TEST_ID = 'user-badge-test-id';
export const USER_BADGE_IMAGE_TEST_ID = 'user-badge-image-test-id';
export const USER_BADGE_NAME_TEST_ID = 'user-badge-name-test-id';
export const TESTIMONIAL_CARD_TEST_ID = 'testimonials-card-test-id';
export const TESTIMONIAL_TEXT_TEST_ID = 'testimonials-text-test-id';
export const LIST_TEST_ID = 'list-test-id';
export const LIST_ITEM_TEST_ID = 'list-item-test-id';
export const SIMPLE_CARD_TEST_ID = 'simple-card-test-id';
export const DIPLOMA_CARD_TEST_ID = 'diploma-card-test-id';
export const DIPLOMA_IMAGE_TEST_ID = 'diploma-image-test-id';
export const DIPLOMA_OVERLAY_TEST_ID = 'diploma-overlay-test-id';
export const PHOTO_CARD_IMAGE_TEST_ID = 'photo-card-image-test-id';
export const PHOTO_CARD_TEXT_TEST_ID = 'photo-card-text-test-id';
export const VIDEO_IFRAME_TEST_ID = 'video-iframe-test-id';
export const VIDEO_PLAY_BUTTON_TEST_ID = 'video-play-button-test-id';
export const VIDEO_TEST_ID = 'video-test-id';
export const VIDEO_THUMBNAIL_TEST_ID = 'video-thumbnail-test-id';
export const FLIP_CARD_BACK_TEST_ID = 'flip-card-back-test-id';
export const FLIP_CARD_FRONT_TEST_ID = 'flip-card-front-test-id';
export const FLIP_CARD_TEST_ID = 'flip-card-test-id';
export const SERVICE_CARD_FRONT_DESCRIPTION_TEST_ID =
  'service-card-front-description-test-id';
export const SERVICE_CARD_FRONT_HEADER_TEST_ID =
  'service-card-front-header-test-id';
export const SERVICE_CARD_BACK_HEADER_TEST_ID =
  'service-card-back-header-test-id';
export const SERVICE_CARD_BACK_LIST_ITEM_TEST_ID =
  'service-card-back-list-item-test-id';
export const SERVICE_CARD_BACK_LIST_TEST_ID = 'service-card-back-list-test-id';
export const SOCIAL_MEDIA_LABEL_TEST_ID = 'social-media-label-test-id';
export const SOCIAL_MEDIA_DESCRIPTION_TEST_ID =
  'social-media-description-test-id';
export const FOOTER_SECTION_TEST_ID = 'footer-section-test-id';
export const PAGINATION_DOT_TEST_ID = 'pagination-dot-test-id';
export const CAROUSEL_NAV_WRAPPER_TEST_ID = 'carousel-nav-wrapper-test-id';
export const CAROUSEL_NAV_NEXT_TEST_ID = 'carousel-nav-next-test-id';
export const CAROUSEL_NAV_PREV_TEST_ID = 'carousel-nav-prev-test-id';
export const ENDLESS_CAROUSEL_TEST_ID = 'endless-carousel-test-id';
export const ENDLESS_CAROUSEL_CONTENT_TEST_ID =
  'endless-carousel-content-test-id';
export const MAIN_SPRAY_BG_TEST_ID = 'main-spray-bg-test-id';
export const HERO_MAIN_CARD_ICON_TEST_ID = 'hero-main-card-icon-test-id';
export const HERO_TEXT_CONTENT_TEST_ID = 'hero-text-content-test-id';
export const HERO_TEXT_CONTENT_ICON_TEST_ID = 'hero-text-content-icon-test-id';
export const EXPERTISE_SECTION_TEST_ID = 'expertise-section-test-id';
export const SCROLL_UP_BUTTON_TEST_ID = 'scroll-up-button-test-id';
export const MODAL_WRAPPER_TEST_ID = 'modal-wrapper-test-id';
export const MODAL_CONTENT_TEST_ID = 'modal-content-test-id';
export const MODAL_CLOSE_BUTTON_TEST_ID = 'modal-close-button-test-id';
export const CASE_MODAL_TEXT_TEST_ID = 'case-modal-text-test-id';
export const TESTIMONIAL_MODAL_NAME_TEST_ID = 'testimonial-modal-name-test-id';
export const TESTIMONIAL_MODAL_POSITION_TEST_ID =
  'testimonial-modal-position-test-id';
export const TESTIMONIAL_MODAL_TEXT_TEST_ID = 'testimonial-modal-text-test-id';
export const TESTIMONIAL_MODAL_IMAGE_TEST_ID =
  'testimonial-modal-image-test-id';
