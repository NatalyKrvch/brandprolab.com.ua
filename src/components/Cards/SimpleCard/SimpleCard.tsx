import clsx from 'clsx';
import type { PropsWithChildren } from 'react';
import { twMerge } from 'tailwind-merge';

import { SIMPLE_CARD_TEST_ID } from '@/lib/testIDs';

import type { SimpleCardProps } from './types';

const SimpleCard = ({
  children,
  borderColor,
  className,
  backgroundImageUrl,
  backgroundColor = 'bg-white',
}: PropsWithChildren<SimpleCardProps>) => {
  return (
    <div
      className={twMerge(
        clsx(
          className,
          backgroundColor,
          borderColor && `border ${borderColor}`,
          'w-full rounded-24 tablet:rounded-32 desktop:rounded-40',
          'bg-cover bg-center',
        ),
      )}
      style={
        backgroundImageUrl
          ? { backgroundImage: `url(${backgroundImageUrl})` }
          : {}
      }
      data-testid={SIMPLE_CARD_TEST_ID}
    >
      {children}
    </div>
  );
};

export default SimpleCard;
