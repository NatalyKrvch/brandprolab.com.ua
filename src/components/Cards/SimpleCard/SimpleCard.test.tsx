import '@testing-library/jest-dom';

import { render, screen } from '@testing-library/react';

import { SIMPLE_CARD_TEST_ID } from '@/lib/testIDs';

import SimpleCard from './SimpleCard';

describe('SimpleCard component', () => {
  const defaultProps = {
    children: <p>Test Content</p>,
    className: 'custom-class',
    backgroundColor: 'bg-gray-100',
  };

  it('renders the SimpleCard component', () => {
    const { asFragment } = render(<SimpleCard {...defaultProps} />);
    const cardElement = screen.getByTestId(SIMPLE_CARD_TEST_ID);

    expect(cardElement).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders children inside the card', () => {
    const { asFragment } = render(<SimpleCard {...defaultProps} />);
    const cardElement = screen.getByTestId(SIMPLE_CARD_TEST_ID);

    expect(cardElement).toHaveTextContent('Test Content');
    expect(asFragment()).toMatchSnapshot();
  });

  it('applies correct className and backgroundColor', () => {
    const { asFragment } = render(<SimpleCard {...defaultProps} />);
    const cardElement = screen.getByTestId(SIMPLE_CARD_TEST_ID);

    expect(cardElement).toHaveClass('custom-class', 'bg-gray-100');
    expect(asFragment()).toMatchSnapshot();
  });

  it('applies background image styles when backgroundImage is provided', () => {
    const backgroundImage = '/test-image.jpg';
    const { asFragment } = render(
      <SimpleCard {...defaultProps} backgroundImageUrl={backgroundImage} />,
    );
    const cardElement = screen.getByTestId(SIMPLE_CARD_TEST_ID);

    expect(cardElement).toHaveStyle(
      `background-image: url(${backgroundImage})`,
    );
    expect(cardElement).toHaveClass('bg-cover', 'bg-center');
    expect(asFragment()).toMatchSnapshot();
  });
});
