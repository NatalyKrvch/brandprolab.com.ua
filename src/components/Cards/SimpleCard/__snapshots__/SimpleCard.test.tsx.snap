// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimpleCard component applies background image styles when backgroundImage is provided 1`] = `
<DocumentFragment>
  <div
    class="custom-class bg-gray-100 w-full rounded-24 tablet:rounded-32 desktop:rounded-40 bg-cover bg-center"
    data-testid="simple-card-test-id"
    style="background-image: url(/test-image.jpg);"
  >
    <p>
      Test Content
    </p>
  </div>
</DocumentFragment>
`;

exports[`SimpleCard component applies correct className and backgroundColor 1`] = `
<DocumentFragment>
  <div
    class="custom-class bg-gray-100 w-full rounded-24 tablet:rounded-32 desktop:rounded-40 bg-cover bg-center"
    data-testid="simple-card-test-id"
  >
    <p>
      Test Content
    </p>
  </div>
</DocumentFragment>
`;

exports[`SimpleCard component renders children inside the card 1`] = `
<DocumentFragment>
  <div
    class="custom-class bg-gray-100 w-full rounded-24 tablet:rounded-32 desktop:rounded-40 bg-cover bg-center"
    data-testid="simple-card-test-id"
  >
    <p>
      Test Content
    </p>
  </div>
</DocumentFragment>
`;

exports[`SimpleCard component renders the SimpleCard component 1`] = `
<DocumentFragment>
  <div
    class="custom-class bg-gray-100 w-full rounded-24 tablet:rounded-32 desktop:rounded-40 bg-cover bg-center"
    data-testid="simple-card-test-id"
  >
    <p>
      Test Content
    </p>
  </div>
</DocumentFragment>
`;
